pkg/identity/userid.go

```diff
@@ -0,0 +1,104 @@
+package identity
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	"github.com/tyler-smith/go-bip39"
+
+	"github.com/threefoldtech/zos/pkg/versioned"
+)
+
+// UserIdentity defines serializable strict to identify a user
+type UserIdentity struct {
+	// Seed encoded in base64
+	Mnemonic string `json:"mnemonic"`
+	// ThreebotID generated by explorer
+	ThreebotID uint64 `json:"threebotid"`
+}
+
+// UserData contains all information needed to identify someone
+type UserData struct {
+	// Seed is the KeyPair object decoded from seed base64
+	Key KeyPair
+	// ThreebotID is a copy from file
+	ThreebotID uint64
+}
+
+// LoadUserIdentity load a UserIdentity from file
+func LoadUserIdentity(path string) (UserData, error) {
+	// Initializing empty object
+	ud := UserData{}
+
+	version, buf, err := versioned.ReadFile(path)
+	if err != nil {
+		return ud, err
+	}
+
+	if version.NE(seedVersionLatest) {
+		return ud, fmt.Errorf("unsupported seed version")
+	}
+
+	var ui UserIdentity
+	err = json.Unmarshal(buf, &ui)
+	if err != nil {
+		return ud, err
+	}
+
+	return LoadUserIdentityObject(&ui)
+}
+
+// LoadUserIdentityObject converts a UserIdentity object to UserData
+func LoadUserIdentityObject(ui *UserIdentity) (UserData, error) {
+	// Initializing empty object
+	ud := UserData{}
+
+	seed, err := bip39.EntropyFromMnemonic(ui.Mnemonic)
+	if err != nil {
+		return ud, err
+	}
+
+	// Loading mnemonic
+	kp, err := FromSeed(seed)
+	if err != nil {
+		return ud, err
+	}
+
+	// Link keys
+	ud.Key = kp
+
+	// Link ThreebotID
+	ud.ThreebotID = ui.ThreebotID
+
+	return ud, nil
+}
+
+// SaveUserData dumps UserData into a file
+func SaveUserData(u *UserData, path string) error {
```

Comment:
Same here.

Reply:
This can be set as a method, right ?

Reply:
Well the UserData struct won't exist anymore so this method has no reason o exists